### PR TITLE
incremental builds v2

### PR DESCRIPTION
**Pull request by ddollar**
uploads the changes as a tarball rather than single files which drastically reduces the overhead

### CODE REVIEW

**Comment by ddollar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;heRH2gih&#x2F;456-incremental-builds-v2"> incremental builds v2</a>

File: api/controllers/index.go [hunk 1]
```diff
@@ -30,28 +32,46 @@ func IndexDiff(rw http.ResponseWriter, r *http.Request) *httperr.Error {
...
+			break
+		}
+		if err != nil {
+			return httperr.Server(err)
+		}
 
-	err = provider.IndexUpload(hash, data)
-	if err != nil {
-		return httperr.Server(err)
+		fmt.Printf("header = %+v\n", header)
```
Comment (MiguelMoll) [inline api/controllers/index.go]: This printed out a lot logs for the rails example. Don't think we really need it.

**Comment by MiguelMoll** [comment]
Getting this output 

```
⟩ convox build --incremental
Analyzing source... OK
Identifying changes... 185 files
Uploading... 105.20 KB / 105.20 KB  100.00 % 22s
Starting build... OK
Release: RFJXGTNTRFA
```

Even though nothing has changed between builds.

**Comment by skwp** [comment]
👏

**Comment by ddollar** [comment]
Could use some help testing this if anyone is feeling up for it.

**Pull request closed without merging**